bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java

```diff
@@ -184,11 +171,14 @@ private void processCommand(PullRequest pr, CensusInstance censusInstance, Path
         if (handler.isPresent()) {
             if (isCommit) {
                 if (handler.get().allowedInCommit()) {
-                    var hash = resultingCommitHash(allComments);
+                    var hash = pr.findIntegratedCommitHash();
```

Comment:
It's worth noting that this will trigger a new `comments()` call to the forge instead of reusing the existing list of comments.

Reply:
Yes, it is a tradeoff between code concision and the running time. If we find the code becomes slower in the future and want to optimize it, this is a good place to optimize. But now we can keep it because we want to refactor the method in this PR.

And the method `pr.findIntegratedCommitHash` always get the newest comments so it may avoid some issues that the `Pushed as ...` comment is added during processing command. But I am not really know whether this situation would happen.